src/core/server/rendering/rendering_service.tsx

```diff
@@ -0,0 +1,126 @@
+    plugins,
+  }: RenderingSetupDeps): Promise<RenderingServiceSetup> {
+    const uiPlugins = await Promise.all(
+      [...plugins.uiPlugins.public].map(async ([pluginId, plugin]) => {
+        const config = ((await plugins.uiPlugins.browserConfigs
+          .get(pluginId)
+          ?.pipe(take(1))
+          .toPromise()) ?? {}) as Record<string, any>;
+
+        return { id: pluginId, plugin, config };
+      })
+    );
+
+    return {
+      getRenderingProvider: ({ request, uiSettings, injectedVarsOverrides }) => {
+        const basePath = http.basePath.get(request);
+
+        return {
+          render: async (
+            id = 'core',
+            options: { includeUserSettings: boolean } = { includeUserSettings: true }
+          ) => {
+            const { env } = this.coreContext;
+            const settings = {
+              defaults: await uiSettings.getAll(),
+              user: options.includeUserSettings ? await uiSettings.getUserProvided() : {},
+            };
+            const vars = await legacy.getVars(
+              id,
+              ensureRawRequest(request),
+              uiPlugins.map(({ config }) => config),
+              injectedVarsOverrides
+            );
+            const metadata = {
+              strictCsp: http.csp.strict,
+              uiPublicUrl: `${basePath}/ui`,
+              bootstrapScriptUrl: `${basePath}/bundles/app/${id}/bootstrap.js`,
+              i18n: i18n.translate,
+              locale: i18n.getLocale(),
+              darkMode: settings.user?.['theme:darkMode']?.userValue
+                ? Boolean(settings.user['theme:darkMode'].userValue)
+                : false,
+              injectedMetadata: {
+                version: env.packageInfo.version,
+                buildNumber: env.packageInfo.buildNum,
+                branch: env.packageInfo.branch,
+                basePath,
+                env,
+                legacyMode: id !== 'core',
+                i18n: {
+                  translationsUrl: `${basePath}/translations/${i18n.getLocale()}.json`,
+                },
+                csp: { warnLegacyBrowsers: http.csp.warnLegacyBrowsers },
+                vars,
+                uiPlugins,
+                legacyMetadata: {
+                  app: {},
+                  bundleId: `app:${id}`,
+                  nav: legacyPlugins.navLinks,
+                  version: env.packageInfo.version,
+                  branch: env.packageInfo.branch,
+                  buildNum: env.packageInfo.buildNum,
+                  buildSha: env.packageInfo.buildSha,
+                  serverName: http.server.name,
+                  devMode: env.mode.dev,
+                  basePath,
+                  uiSettings: settings,
+                },
+              },
+            };
+
+            const markup = `<!DOCTYPE html>${renderToStaticMarkup(
+              <Template metadata={metadata} />
+            )}`;
+
+            return format(markup, { parser: 'html' });
```

Comment:
Github won't allow me to comment on the [original thread](https://github.com/elastic/kibana/pull/52161#discussion_r353799578) anymore, so starting another one here.

Arguments for doing this were mostly about consistent output for testing. I don't see why it wouldn't be consistent without this, at least between test runs. I just don't think the downsides here are worth the benefit:
- Shipping dev tooling in production
- Potential breaking if prettier throws an error when parsing
- Potential security risks when parsing that leads to an RCE vulnerability
  - This is theoretical risk, but anytime we're doing parsing in JavaScript I get nervous, especially with a library that is not usually used in production environments.
- Small CPU overhead, small network overhead
  - These are probably both negiglble